src/Command/AssertBackwardsCompatible.php

```diff
@@ -115,9 +115,27 @@ public function execute(InputInterface $input, OutputInterface $output): int
         // @todo fix flaky assumption about the path of the source repo...
         $sourceRepo = CheckedOutRepository::fromPath(Env\current_dir());
 
-        $fromRevision = $input->getOption('from') !== null
-            ? $this->parseRevisionFromInput($input, $sourceRepo)
-            : $this->determineFromRevisionFromRepository($sourceRepo, $stdErr);
+        if ($input->getOption('from') !== null) {
+            $maybeRevision = Type\string()->coerce($input->getOption('from'));
+        } else {
+            $versions = $this->getVersions->fromRepository($sourceRepo);
+
+            Psl\invariant(Iter\count($versions) >= 1, 'Could not detect any released versions for the given repository');
+
+            $stableVersions = $versions->matching(new StableVersionConstraint());
+
+            if ($stableVersions->isEmpty()) {
+                $stdErr->writeln(Str\format('Your library does not have any stable versions. Therefore cannot have BC breaks.'));
+
+                return Command::SUCCESS;
```

Comment:
I just realized another problem, sorry @_@ 

Assume somebody incorrectly installs this tool with a CI setup that did **not** pull tags during clone operations (shallow clone: default on github), this leads to an exit code zero by default, hiding the underlying problem.

Perhaps an exception or non-zero exit code was really the best approach.

Reply:
Actually, maybe we can kill two birds with one stone. One is a misconfiguration which we could verify and the second is an exception :)

Reply:
>  One is a misconfiguration which we could verify

How do we verify it, if no `git fetch` has occurred? :thinking:

Reply:
Isn't that...

No tags -> possible misconfiguration so we can throw an exception. Like. THERE ARE NO TAGS. WE NEED THEM (yes, we gonna scream 😄)

Then at some point, we can assume that there are at least some tags, but we want to check out the stable ones. There are tags, but none of them are stable. Return success.

What do you think?